### ISSUE TITLE
linprog answer differs when bound is added to constraints

### ISSUE THREAD

**Issue by savrus** [scipy/scipy#8662]
optmize.linprog produces wrong answer with default bounds (which are (0, None)) However the answer is correct if the boundary condition is moved to constraints.

### Reproducing code example:
```
import scipy.optimize

c = [-10, 10, 6, 3]
A= [
    [8, -8, -4, 6],
    [-8, 8, 4, -6],
    [-4, 4, 8, -4],
    [3, -3, -3, -10]
]
b = [9, -9, -9, -4]
bounds = [(0, None), (0, None), (0, None), (0, None)]

res1 = scipy.optimize.linprog(c, A, b, bounds=bounds)
print res1
print

A.append([0, 0, -1, 0])
b.append(0)
bounds[2] = (None, None)

res2 = scipy.optimize.linprog(c, A, b, bounds=bounds)
print res2
print

assert not (res1.success and res2.success) or abs(res1.fun - res2.fun) < 1
```

### Error message:
```
     fun: 100.00000000000004
 message: 'Optimization terminated successfully.'
     nit: 5
   slack: array([   0.  ,    0.  ,    0.  ,  122.75])
  status: 0
 success: True
       x: array([  0.  ,   6.25,   1.  ,  10.5 ])

     fun: 36.000000000000028
 message: 'Optimization terminated successfully.'
     nit: 7
   slack: array([  0.00000000e+00,  -8.88178420e-16,   0.00000000e+00,
         4.77500000e+01,   0.00000000e+00])
  status: 0
 success: True
       x: array([ 0.  ,  2.25,  0.  ,  4.5 ])

Traceback (most recent call last):
  File "buglinprog.py", line 25, in <module>
    assert not (res1.success and res2.success) or abs(res1.fun - res2.fun) < 1
AssertionError
```

### Scipy/Numpy/Python version information:
```
('1.0.1', '1.8.2', sys.version_info(major=2, minor=7, micro=6, releaselevel='final', serial=0))
```

**Comment by mdhaber** [comment]
I don't think I can help with this one. Try interior-point.

**Comment by Kai-Striega** [comment]
I've attempted to diagnose and fix this bug using:

`Anaconda 4.5.4, SciPy 1.1.0 and Python 3.5.4.`

Firstly I can reproduce the incorrect result given by the default constraints.

**Input**

```
import scipy.optimize
c = [-10, 10, 6, 3]
A= [
    [8, -8, -4, 6],
    [-8, 8, 4, -6],
    [-4, 4, 8, -4],
    [3, -3, -3, -10]
]
b = [9, -9, -9, -4]
bounds = [(0, None), (0, None), (0, None), (0, None)]
res = scipy.optimize.linprog(
    c, A, b,
    bounds=bounds)
print(res)
```

**output**
```
 fun: 100.00000000000004
 message: 'Optimization terminated successfully.'
     nit: 5
   slack: array([   0.  ,    0.  ,    0.  ,  122.75])
  status: 0
 success: True
       x: array([  0.  ,   6.25,   1.  ,  10.5 ])
```

The error first appears in the simplex tableau during the first iteration of phase 2, in the form of extremely large coefficients, which can be seen using the `callback` argument.
**input:**
```
res = scipy.optimize.linprog(
    c, A, b,
    bounds=bounds,
    callback=scipy.optimize.linprog_verbose_callback)
```
**output:**
```
...non-relevant output...

--------- Iteration Complete - Phase 2 -------

Tableau:
[[      0.0000       0.0000       0.0000       0.0000 -84442493013196816.0000 -84442493013196816.0000     -12.2500       1.0000     122.7500]
 [      0.0000       0.0000       1.0000       0.0000 -1125899906842624.0000 -1125899906842624.0000       0.0000       0.0000       1.0000]
 [     -1.0000       1.0000       0.0000       0.0000 -4503599627370498.0000 -4503599627370497.0000      -0.7500       0.0000       6.2500]
 [      0.0000       0.0000       0.0000       1.0000 -6755399441055746.0000 -6755399441055745.0000      -1.0000       0.0000      10.5000]
 [      0.0000       0.0000       0.0000       0.0000  72057594037927952.0000  72057594037927952.0000      10.5000       0.0000    -100.0000]]

Basic Variables: [7 2 1 3]

Current Solution:
x =  [      0.0000       6.2500       1.0000      10.5000]

Current Objective Value:
f =  100.00000000000004

...output of following steps...
```
These large values occur during the check for remaining artificial variables. If any artificial variables are found they are removed with another pivot. The column in which to pivot corresponds that of a non-artificial variable having a non-zero coefficient. If the coefficients is very close to 0 (in this example on the order of 1e-18) the pivot results in division by a very small value producing the errors.

I have changed the predicate to test if the coefficient lies outside some tolerance level, using the already defined `tol` variable. 

**From:**
```
for pivrow in [row for row in range(basis.size)
                       if basis[row] > T.shape[1] - 2]:
            non_zero_row = [col for col in range(T.shape[1] - 1) 
                            if T[pivrow, col] != 0]
```

**To: **
```
for pivrow in [row for row in range(basis.size)
                       if basis[row] > T.shape[1] - 2]:
            non_zero_row = [col for col in range(T.shape[1] - 1)
                            if abs(T[pivrow, col]) > tol]

```
This *may* still lead to some errors as the default value of `tol=1.0e-12 ` , however it is still better by a factor of a million. The change now produces equivalent results for both problem formulations.

**Comment by mdhaber** [comment]
Closed via #8909
